conans/errors.py

```diff
@@ -102,6 +102,16 @@ def __str__(self):
         return exception_message_safe(msg)
 
 
+class ConanReferenceDoesNotExist(ConanException):
```

Comment:
If these are DB/Cache errors, maybe add DB/Cache to the name itself? It could read as a 404, server side missing package error.